packages/date-fns-jalali/src/date-fns-utils.ts

```diff
@@ -0,0 +1,323 @@
+import eachDayOfInterval from "date-fns-jalali/eachDayOfInterval";
+import endOfDay from "date-fns-jalali/endOfDay";
+import endOfWeek from "date-fns-jalali/endOfWeek";
+import endOfYear from "date-fns-jalali/endOfYear";
+import format from "date-fns-jalali/format";
+import getHours from "date-fns-jalali/getHours";
+import getSeconds from "date-fns-jalali/getSeconds";
+import getYear from "date-fns-jalali/getYear";
+import getMonth from "date-fns-jalali/getMonth";
+import getMinutes from "date-fns-jalali/getMinutes";
+import isAfter from "date-fns-jalali/isAfter";
+import isBefore from "date-fns-jalali/isBefore";
+import isEqual from "date-fns-jalali/isEqual";
+import isSameDay from "date-fns-jalali/isSameDay";
+import isSameYear from "date-fns-jalali/isSameYear";
+import isSameMonth from "date-fns-jalali/isSameMonth";
+import isSameHour from "date-fns-jalali/isSameHour";
+import isValid from "date-fns-jalali/isValid";
+import dateFnsParse from "date-fns-jalali/parse";
+import setHours from "date-fns-jalali/setHours";
+import setMinutes from "date-fns-jalali/setMinutes";
+import setMonth from "date-fns-jalali/setMonth";
+import setSeconds from "date-fns-jalali/setSeconds";
+import setYear from "date-fns-jalali/setYear";
+import startOfDay from "date-fns-jalali/startOfDay";
+import startOfMonth from "date-fns-jalali/startOfMonth";
+import endOfMonth from "date-fns-jalali/endOfMonth";
+import startOfWeek from "date-fns-jalali/startOfWeek";
+import startOfYear from "date-fns-jalali/startOfYear";
+
+// Locale is not exported from date-fns, so we need to workaround that https://github.com/date-fns/date-fns/issues/932
+import SampleLocale from "date-fns-jalali/locale/fa-jalali-IR";
+import { IUtils } from "@date-io/core/IUtils";
+
+type Locale = typeof SampleLocale;
+
+export default class DateFnsJalaliUtils implements IUtils<Date> {
+  public locale?: Locale;
+
+  public yearFormat = "yyyy";
+
+  public yearMonthFormat = "MMMM yyyy";
+
+  public dateTime12hFormat = "yyyy/MM/dd hh:mm aaaa";
+
+  public dateTime24hFormat = "yyyy/MM/dd HH:mm";
+
+  public time12hFormat = "hh:mm a";
+
+  public time24hFormat = "HH:mm";
+
+  public dateFormat = "yyyy/MM/dd";
+
+  constructor({ locale }: { locale?: Locale } = {}) {
+    this.locale = locale;
+  }
+
+  // Note: date-fns input types are more lenient than this adapter, so we need to expose our more
+  // strict signature and delegate to the more lenient signature. Otherwise, we have downstream type errors upon usage.
+
+  public addDays(value: Date, count: number) {
+    return addDays(value, count);
+  }
+
+  public isValid(value: any) {
+    return isValid(this.date(value));
+  }
+
+  public getDiff(value: Date, comparing: Date | string) {
+    return differenceInMilliseconds(value, this.date(comparing));
+  }
+
+  public isAfter(value: Date, comparing: Date) {
+    return isAfter(value, comparing);
+  }
+
+  public isBefore(value: Date, comparing: Date) {
+    return isBefore(value, comparing);
```

Comment:
Do calculation methos are difffernt for jalaali adapter?

Reply:
note: `date-fns` is not dependency of `date-io/date-fns-jalali`